profile/src/main/resources/profileschema/datahelix.schema.json

```diff
@@ -947,7 +954,67 @@
           "type": "string"
         }
       }
+    },
+    "relationship": {
+      "title": "Defines any nested child objects",
+      "type": "object",
+      "additionalProperties": false,
+      "required": [
+        "name"
+      ],
+      "default": {
+        "name": "relationshipName",
+        "type": "string",
+        "formatting": "",
+        "unique": false,
+        "nullable": true
+      },
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        },
+        "profileFile": {
+          "type": "string"
+        },
+        "profile": {
+          "type": "object",
+          "additionalProperties": false,
```

Comment:
Can you 'ref' back to the main schema here? This object should confirm to the profile schema, and can contain relationships itself which isn't represented here?

Reply:
Ah, hadn't realised a relationship could contain another relationship.  I can't work out how to refer back to the main schema, but I have added a relationships section at the bottom, which seems to work.  Not ideal to have fields, constraints and relationships defined twice though.

Reply:
Can you do something like this? 

```
...
"profile": {
   "$ref": "#/"
}
...
```

Reply:
I don't think so.  From how I understand it you have to have some initial entries in the schema first, then they can refer to definitions below.  I don't think you can effectively go straight to the definitions.  I tried the above anyhow but it didn't work for me.

Reply:
As discussed @sl-slaing, our aim is to not change the profile format itself and to fix this in the schema.  We also don't want to have fields, constraints and relationships defined in two places.  The [Recursion](https://json-schema.org/understanding-json-schema/structuring.html#id2) section of the JSON Schema Reference should help, just can't figure it out at the moment.